previewer/preview_renderer.js

```diff
@@ -4,20 +4,58 @@ $ = require('jquery')
 assetMap = main.assetMap
 hatchFolder = main.hatchFolder
 
+showingSource = false;
+
 $(document).ready(function(){
+  $('#flip_controls').hide();
+  $('#source_code').hide();
 })
 
+flipPage = function() {
+  if (showingSource) {
+    $('#preview').hide();
+    $('#source_code').show();
+  } else {
+    $('#source_code').hide();
+    $('#preview').show();
+  }
+  showingSource = !showingSource;
+}
+
 setAssetID = function(ID) {
-  if (ID != null) {
-    var asset = assetMap.get(ID)
-    console.log(asset)
-    $('#preview').html(asset.document)
+  if (!ID) {
+    $('#preview').html("<center><h2>No asset selected</h2></center>")
+    return;
+  }
+
+  showingSource = false;
+  $('#source_code').hide();
+  $('#preview').show();
+  $('#image_asset').remove();
+
+  var asset = assetMap.get(ID)
+
+  // Enable controls if we're an html document
+  if (asset.document) {
+    $('#flip_controls').show();
+    $('#source_code_content').text(asset.document).html();
+  } else {
+    $('#flip_controls').hide();
+  }
+
+  $('#preview').html(asset.document || "")
```

Comment:
@wgreenberg This one is just to make sure we start with a clean canvas and can handle `undefine`s. The code below here will tack on an image to the preview tag too so having any text there wouldn't work.